From 0ab2622c947396a43cf325324d693898751b126e Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Wed, 13 Jul 2022 14:21:29 +0200
Subject: [PATCH] meson: remove deprecated positional arguments from
 i18n.merge_file calls

---
 data/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/data/meson.build b/data/meson.build
index 77a7909..09feaa7 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -8,7 +8,6 @@ i18n.merge_file (
 )
 
 i18n.merge_file (
-    'desktop',
     input: 'shortcut-overlay.desktop.in',
     output: meson.project_name() + '.desktop',
     po_dir: join_paths(meson.source_root (), 'po', 'extra'),
-- 
2.37.1
