diff -Naur ffmpeg-7.1.2.old/libavcodec/lcevcdec.c ffmpeg-7.1.2/libavcodec/lcevcdec.c
--- ffmpeg-7.1.2.old/libavcodec/lcevcdec.c	2025-10-27 10:07:00.369472051 +0100
+++ ffmpeg-7.1.2/libavcodec/lcevcdec.c	2025-10-27 10:07:01.223246583 +0100
@@ -139,7 +139,7 @@
     if (!sd)
         return 1;
 
-    res = LCEVC_SendDecoderEnhancementData(lcevc->decoder, in->pts, 0, sd->data, sd->size);
+    res = LCEVC_SendDecoderEnhancementData(lcevc->decoder, (uint64_t) in->pts, sd->data, sd->size);
     if (res != LCEVC_Success)
         return AVERROR_EXTERNAL;
 
@@ -147,7 +147,7 @@
     if (ret < 0)
         return ret;
 
-    res = LCEVC_SendDecoderBase(lcevc->decoder, in->pts, 0, picture, -1, NULL);
+    res = LCEVC_SendDecoderBase(lcevc->decoder, (uint64_t) in->pts, picture, -1, NULL);
     if (res != LCEVC_Success)
         return AVERROR_EXTERNAL;
 
diff -Naur ffmpeg-7.1.2.old/libavfilter/vf_lcevc.c ffmpeg-7.1.2/libavfilter/vf_lcevc.c
--- ffmpeg-7.1.2.old/libavfilter/vf_lcevc.c	2025-10-27 10:07:00.732521928 +0100
+++ ffmpeg-7.1.2/libavfilter/vf_lcevc.c	2025-10-27 10:07:01.223846623 +0100
@@ -139,7 +139,7 @@
         return ret;
 
     if (sd) {
-        res = LCEVC_SendDecoderEnhancementData(lcevc->decoder, in->pts, 0, sd->data, sd->size);
+        res = LCEVC_SendDecoderEnhancementData(lcevc->decoder, (uint64_t) in->pts, sd->data, sd->size);
         if (res == LCEVC_Again)
             return AVERROR(EAGAIN);
         else if (res != LCEVC_Success) {
@@ -148,7 +148,7 @@
         }
     }
 
-    res = LCEVC_SendDecoderBase(lcevc->decoder, in->pts, 0, picture, -1, in);
+    res = LCEVC_SendDecoderBase(lcevc->decoder, (uint64_t) in->pts, picture, -1, in);
     if (res != LCEVC_Success) {
         av_log(ctx, AV_LOG_ERROR, "LCEVC_SendDecoderBase failed\n");
         LCEVC_FreePicture(lcevc->decoder, picture);
