diff --git a/Cargo.toml b/Cargo.toml
index 4e3f31fc39c3340e29c842c00f7966a18ff80bd5..f0d64e58fc683d3421c9d9a0cd5753c5e2873fe5 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -79,7 +79,6 @@ regex = "1"
 remove_dir_all = { version = "1.0.0", features = ["parallel"] }
 reqwest = { version = "0.13", default-features = false, features = ["blocking", "gzip", "http2", "socks", "stream"], optional = true }
 retry = { version = "2", default-features = false, features = ["random"] }
-rs_tracing = { version = "1.1", features = ["rs_tracing"] }
 rustls = { version = "0.23", optional = true, default-features = false, features = ["logging", "aws_lc_rs", "tls12"] }
 rustls-platform-verifier = { version = "0.7", optional = true }
 same-file = "1"
diff --git a/src/bin/rustup-init.rs b/src/bin/rustup-init.rs
index ded74b236a1401408ef6e72b467261c83767a160..9c404b1dca95e5ee78965a5cf7c95d03a9c60bc1 100644
--- a/src/bin/rustup-init.rs
+++ b/src/bin/rustup-init.rs
@@ -16,10 +16,6 @@
 use std::process::ExitCode;
 
 use anyhow::{Context, Result, anyhow};
-// Public macros require availability of the internal symbols
-use rs_tracing::{
-    close_trace_file, close_trace_file_internal, open_trace_file, trace_to_file_internal,
-};
 use tracing_subscriber::{EnvFilter, Registry, reload::Handle};
 
 use rustup::cli::errors::CliError;
@@ -66,13 +62,7 @@ async fn run_rustup(
     process: &Process,
     console_filter: Handle<EnvFilter, Registry>,
 ) -> Result<utils::ExitCode> {
-    if let Ok(dir) = process.var("RUSTUP_TRACE_DIR") {
-        open_trace_file!(dir)?;
-    }
     let result = run_rustup_inner(process, console_filter).await;
-    if process.var("RUSTUP_TRACE_DIR").is_ok() {
-        close_trace_file!();
-    }
     result
 }
 
diff --git a/src/diskio/immediate.rs b/src/diskio/immediate.rs
index 779acd3452e270f1d7a921aab9181d6f15192e27..94247ef374168042e41c7be61d125c20953e554e 100644
--- a/src/diskio/immediate.rs
+++ b/src/diskio/immediate.rs
@@ -158,7 +158,6 @@ impl IncrementalFileWriter {
         let path = path.as_ref();
         let path_display = format!("{}", path.display());
         let file = Some({
-            trace_scoped!("creat", "name": path_display);
             opts.write(true).create(true).truncate(true).open(path)?
         });
         Ok(Self {
@@ -200,11 +199,9 @@ impl IncrementalFileWriter {
         };
         // Length 0 vector is used for clean EOF signalling.
         if chunk.is_empty() {
-            trace_scoped!("close", "name:": self.path_display);
             drop(std::mem::take(&mut self.file));
             state.finished = true;
         } else {
-            trace_scoped!("write_segment", "name": self.path_display, "len": chunk.len());
             file.write_all(&chunk)?;
 
             state.completed_chunks.push(chunk.len());
diff --git a/src/diskio/mod.rs b/src/diskio/mod.rs
index 5599a9720c6830e2f8487951f7937e07a40e263b..bd47723639ef52cd8aef0ecca86488474ce01aad 100644
--- a/src/diskio/mod.rs
+++ b/src/diskio/mod.rs
@@ -380,17 +380,14 @@ fn write_file<P: AsRef<Path>, C: AsRef<[u8]>>(path: P, contents: C, mode: u32) -
     let path = path.as_ref();
     let path_display = format!("{}", path.display());
     let mut f = {
-        trace_scoped!("creat", "name": path_display);
         opts.write(true).create(true).truncate(true).open(path)?
     };
     let contents = contents.as_ref();
     let len = contents.len();
     {
-        trace_scoped!("write", "name": path_display, "len": len);
         f.write_all(contents)?;
     }
     {
-        trace_scoped!("close", "name:": path_display);
         drop(f);
     }
     Ok(())
@@ -412,7 +409,6 @@ fn write_file_incremental<P: AsRef<Path>, F: Fn(usize)>(
     let path = path.as_ref();
     let path_display = format!("{}", path.display());
     let mut f = {
-        trace_scoped!("creat", "name": path_display);
         opts.write(true).create(true).truncate(true).open(path)?
     };
     if let IncrementalFile::ThreadedReceiver(recv) = content_callback {
@@ -423,12 +419,10 @@ fn write_file_incremental<P: AsRef<Path>, F: Fn(usize)>(
             let len = contents.len();
             // Length 0 vector is used for clean EOF signalling.
             if len == 0 {
-                trace_scoped!("EOF_chunk", "name": path_display, "len": len);
                 drop(contents);
                 chunk_complete_callback(len);
                 break;
             } else {
-                trace_scoped!("write_segment", "name": path_display, "len": len);
                 f.write_all(&contents)?;
                 drop(contents);
                 chunk_complete_callback(len);
@@ -438,7 +432,6 @@ fn write_file_incremental<P: AsRef<Path>, F: Fn(usize)>(
         unreachable!();
     }
     {
-        trace_scoped!("close", "name:": path_display);
         drop(f);
     }
     Ok(())
@@ -447,7 +440,6 @@ fn write_file_incremental<P: AsRef<Path>, F: Fn(usize)>(
 fn create_dir<P: AsRef<Path>>(path: P) -> io::Result<()> {
     let path = path.as_ref();
     let path_display = format!("{}", path.display());
-    trace_scoped!("create_dir", "name": path_display);
     std::fs::create_dir(path)
 }
 
diff --git a/src/lib.rs b/src/lib.rs
index 0daf2f6eded502ad7479329466d458402fbbb556..3a1e6d3502f3592c9ed8c84e647b6d8dca029539 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -9,9 +9,6 @@ use anyhow::{Result, anyhow};
 use errors::RustupError;
 use itertools::{Itertools, chain};
 
-#[macro_use]
-extern crate rs_tracing;
-
 // A list of all binaries which Rustup will proxy.
 pub static TOOLS: &[&str] = &[
     "rustc",
