From 6ebd57673b45cc64e1caf895134efc0d5f6cf2be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Danielle=20For=C3=A9?= <daniel@elementary.io>
Date: Sat, 23 Apr 2022 15:34:16 -0700
Subject: [PATCH] Read screenshot shortcuts from gala schema (#407)

---
 src/Shortcuts/List.vala | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/Shortcuts/List.vala b/src/Shortcuts/List.vala
index 34a518c9..4d534a16 100644
--- a/src/Shortcuts/List.vala
+++ b/src/Shortcuts/List.vala
@@ -96,12 +96,12 @@ namespace Pantheon.Keyboard.Shortcuts {
             screenshot_group = {};
             screenshot_group.icon_name = "io.elementary.switchboard.keyboard.screenshots";
             screenshot_group.label = _("Screenshots");
-            add_action (ref screenshot_group, Schema.MEDIA, _("Grab the whole screen"), "screenshot");
-            add_action (ref screenshot_group, Schema.MEDIA, _("Copy the whole screen to clipboard"), "screenshot-clip");
-            add_action (ref screenshot_group, Schema.MEDIA, _("Grab the current window"), "window-screenshot");
-            add_action (ref screenshot_group, Schema.MEDIA, _("Copy the current window to clipboard"), "window-screenshot-clip");
-            add_action (ref screenshot_group, Schema.MEDIA, _("Select an area to grab"), "area-screenshot");
-            add_action (ref screenshot_group, Schema.MEDIA, _("Copy an area to clipboard"), "area-screenshot-clip");
+            add_action (ref screenshot_group, Schema.GALA, _("Grab the whole screen"), "screenshot");
+            add_action (ref screenshot_group, Schema.GALA, _("Copy the whole screen to clipboard"), "screenshot-clip");
+            add_action (ref screenshot_group, Schema.GALA, _("Grab the current window"), "window-screenshot");
+            add_action (ref screenshot_group, Schema.GALA, _("Copy the current window to clipboard"), "window-screenshot-clip");
+            add_action (ref screenshot_group, Schema.GALA, _("Select an area to grab"), "area-screenshot");
+            add_action (ref screenshot_group, Schema.GALA, _("Copy an area to clipboard"), "area-screenshot-clip");
 
             launchers_group = {};
             launchers_group.icon_name = "preferences-desktop-applications";
