--- a/source/frontend/carla_shared.py
+++ b/source/frontend/carla_shared.py
@@ -473,30 +473,37 @@ else:
 
     DEFAULT_LADSPA_PATH  = HOME + "/.ladspa"
     DEFAULT_LADSPA_PATH += ":/usr/lib/ladspa"
+    DEFAULT_LADSPA_PATH += ":/usr/lib64/ladspa"
     DEFAULT_LADSPA_PATH += ":/usr/local/lib/ladspa"
 
     DEFAULT_DSSI_PATH    = HOME + "/.dssi"
     DEFAULT_DSSI_PATH   += ":/usr/lib/dssi"
+    DEFAULT_DSSI_PATH   += ":/usr/lib64/dssi"
     DEFAULT_DSSI_PATH   += ":/usr/local/lib/dssi"
 
     DEFAULT_LV2_PATH     = HOME + "/.lv2"
     DEFAULT_LV2_PATH    += ":/usr/lib/lv2"
+    DEFAULT_LV2_PATH    += ":/usr/lib64/lv2"
     DEFAULT_LV2_PATH    += ":/usr/local/lib/lv2"
 
     DEFAULT_VST2_PATH    = HOME + "/.vst"
     DEFAULT_VST2_PATH   += ":/usr/lib/vst"
+    DEFAULT_VST2_PATH   += ":/usr/lib64/vst"
     DEFAULT_VST2_PATH   += ":/usr/local/lib/vst"
 
     DEFAULT_VST2_PATH   += HOME + "/.lxvst"
     DEFAULT_VST2_PATH   += ":/usr/lib/lxvst"
+    DEFAULT_VST2_PATH   += ":/usr/lib64/lxvst"
     DEFAULT_VST2_PATH   += ":/usr/local/lib/lxvst"
 
     DEFAULT_VST3_PATH    = HOME + "/.vst3"
     DEFAULT_VST3_PATH   += ":/usr/lib/vst3"
+    DEFAULT_VST3_PATH   += ":/usr/lib64/vst3"
     DEFAULT_VST3_PATH   += ":/usr/local/lib/vst3"
 
     DEFAULT_CLAP_PATH    = HOME + "/.clap"
     DEFAULT_CLAP_PATH   += ":/usr/lib/clap"
+    DEFAULT_CLAP_PATH   += ":/usr/lib64/clap"
     DEFAULT_CLAP_PATH   += ":/usr/local/lib/clap"
 
     DEFAULT_SF2_PATH     = HOME + "/.sounds/sf2"
