From 4f11ff169ab50537d7ae1c0811090bd0933a64c9 Mon Sep 17 00:00:00 2001
From: windowsboy111 <wboy111@outlook.com>
Date: Sun, 19 Mar 2023 20:10:45 +0800
Subject: [PATCH] fix missing import

---
 src/libappimage/utils/hashlib.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libappimage/utils/hashlib.h b/src/libappimage/utils/hashlib.h
index 45240f7..90c2748 100644
--- a/src/libappimage/utils/hashlib.h
+++ b/src/libappimage/utils/hashlib.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 // system
 #include <istream>
 #include <vector>
-- 
2.39.2

