diff --git a/Makefile b/Makefile
index 34e5952..82be7b3 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@ $(info debug is $(debug))
 
 ifdef debug
 	release :=
-	target :=debug
+	target :=rpm
 	extension :=-debug
 else
 	release :=--release
@@ -19,9 +19,8 @@ install:
 	install -Dm0755 target/$(target)/cardwired /usr/bin/cardwired$(extension)
 	install -Dm0644 assets/cardwired.service /usr/lib/systemd/system/cardwired.service
 	install -Dm0644 assets/com.github.opengamingcollective.cardwire.conf /usr/share/dbus-1/system.d/com.github.opengamingcollective.cardwire.conf
-	systemctl enable cardwired.service
 
 check:
 	cargo clippy --all-targets --all-features -- -D warnings
 
-.PHONY: build install check
\ No newline at end of file
+.PHONY: build install check
