--- a/gtk/module.rules
+++ b/gtk/module.rules
@@ -21,6 +21,7 @@ $(GTK.CONFIGURE.stamp): $(GTK.src/)configure.ac $(GTK.src/)src/Makefile.am
 		CXX="$(GCC.gxx)" \
 		CFLAGS="$(call fn.ARGS,GTK.GCC,.g .O *D *W ?extra)" \
 		LDFLAGS="$(call fn.ARGS,GTK.GCC,?strip .g .O ?extra.exe)" \
+		STRIP="$(STRIP.exe)" \
 		PYTHON="$(PYTHON.exe)" \
 		--prefix=$(PREFIX) \
 		--with-hb=$(call fn.ABSOLUTE,$(BUILD/))
