From 756c5eca8dc764ac194afb7403e75599c3fb4daa Mon Sep 17 00:00:00 2001
From: Raboneko <raboneko@fyralabs.com>
Date: Fri, 8 Sep 2023 15:11:58 +0000
Subject: [PATCH] bump: neovide

---
 anda/rust/neovide/neovide.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/anda/rust/neovide/neovide.spec b/anda/rust/neovide/neovide.spec
index d8858ade..a60536ed 100644
--- a/anda/rust/neovide/neovide.spec
+++ b/anda/rust/neovide/neovide.spec
@@ -1,7 +1,7 @@
 %define debug_package %nil
 
 Name:		neovide
-Version:	0.11.1
+Version:	0.11.2
 Release:	1%{?dist}
 Summary:	No Nonsense Neovim Client in Rust
 License:	MIT
-- 
2.41.0

