From 9ab89e0f57292e282519bc3efeef7e84d7f824aa Mon Sep 17 00:00:00 2001
From: Raboneko <raboneko@fyralabs.com>
Date: Sat, 16 Sep 2023 02:36:12 +0000
Subject: [PATCH] bump: wingpanel-indicator-datetime

---
 .../wingpanel-indicator-datetime.spec                         | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/anda/desktops/elementary/wingpanel-indicator-datetime/wingpanel-indicator-datetime.spec b/anda/desktops/elementary/wingpanel-indicator-datetime/wingpanel-indicator-datetime.spec
index 3286d363..ba4f3f7a 100644
--- a/anda/desktops/elementary/wingpanel-indicator-datetime/wingpanel-indicator-datetime.spec
+++ b/anda/desktops/elementary/wingpanel-indicator-datetime/wingpanel-indicator-datetime.spec
@@ -5,8 +5,8 @@
 
 Name:           wingpanel-indicator-datetime
 Summary:        Datetime Indicator for wingpanel
-Version:        2.4.1
-Release:        2%{?dist}
+Version:        2.4.0
+Release:        1%{?dist}
 License:        GPL-3.0-or-later
 
 URL:            https://github.com/elementary/wingpanel-indicator-datetime
-- 
2.41.0

