From f15e78cd7c2f4243f6c5d7d64da0089d144b0268 Mon Sep 17 00:00:00 2001
From: Raboneko <raboneko@fyralabs.com>
Date: Sat, 23 Sep 2023 06:17:02 +0000
Subject: [PATCH] bump: compiz9

---
 anda/desktops/compiz9/compiz9.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/anda/desktops/compiz9/compiz9.spec b/anda/desktops/compiz9/compiz9.spec
index 99a67d7b..fb3d612a 100644
--- a/anda/desktops/compiz9/compiz9.spec
+++ b/anda/desktops/compiz9/compiz9.spec
@@ -1,4 +1,4 @@
-%define _ubuntu_rel 22.10.20220822-0ubuntu3
+%define _ubuntu_rel 22.10.20220822-0ubuntu4
 %global _hardened_build 0
 
 Name:           compiz9
-- 
2.41.0

