From 00f1bff8cfb846830eb5623349f96d34f5f66fac Mon Sep 17 00:00:00 2001
From: Raboneko <raboneko@fyralabs.com>
Date: Fri, 22 Sep 2023 21:36:28 +0000
Subject: [PATCH] bump: fuc

---
 anda/tools/fuc/fuc.spec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/anda/tools/fuc/fuc.spec b/anda/tools/fuc/fuc.spec
index aada8675..580b3956 100644
--- a/anda/tools/fuc/fuc.spec
+++ b/anda/tools/fuc/fuc.spec
@@ -1,8 +1,8 @@
 %global debug_package %{nil}
 
 Name:			fuc
-Version:		1.1.7
-Release:		2%{?dist}
+Version:		1.1.8
+Release:		1%{?dist}
 Summary:		Modern, performance focused unix commands
 URL:			https://github.com/SUPERCILEX/fuc
 Source0:		https://raw.githubusercontent.com/SUPERCILEX/fuc/%{version}/README.md
-- 
2.41.0

