From c285421f59f78535b3a6fb8e3297d85cb2cccf33 Mon Sep 17 00:00:00 2001
From: Raboneko <raboneko@fyralabs.com>
Date: Fri, 15 Sep 2023 14:12:45 +0000
Subject: [PATCH] bump: sops

---
 anda/tools/sops/sops.spec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/anda/tools/sops/sops.spec b/anda/tools/sops/sops.spec
index 9c55ac52..9afd9178 100644
--- a/anda/tools/sops/sops.spec
+++ b/anda/tools/sops/sops.spec
@@ -1,7 +1,7 @@
 %define debug_package %nil
 Name:			sops
-Version:		3.7.3
-Release:		2%?dist
+Version:		3.8.0
+Release:		1%{?dist}
 Summary:		Simple and flexible tool for managing secrets
 License:		MPL-2.0
 URL:			https://github.com/getsops/sops
-- 
2.41.0

