From 219207477dd3e2cab717b821004b79a4e90cba5a Mon Sep 17 00:00:00 2001
From: Raboneko <raboneko@fyralabs.com>
Date: Fri, 22 Sep 2023 11:11:54 +0000
Subject: [PATCH] bump: libhelium

---
 anda/lib/libhelium/libhelium.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/anda/lib/libhelium/libhelium.spec b/anda/lib/libhelium/libhelium.spec
index 24c92a29..c7750095 100644
--- a/anda/lib/libhelium/libhelium.spec
+++ b/anda/lib/libhelium/libhelium.spec
@@ -1,6 +1,6 @@
 Summary:        The Application Framework for tauOS apps
 Name:           libhelium
-Version:        1.7.6
+Version:        1.8.0
 Release:        1%{?dist}
 License:        GPL-3.0
 URL:            https://github.com/tau-OS/libhelium
-- 
2.41.0

