From 7081a139658ce5cfab0e587631acfcba17abb1d9 Mon Sep 17 00:00:00 2001
From: Raboneko <raboneko@fyralabs.com>
Date: Tue, 19 Sep 2023 12:27:56 +0000
Subject: [PATCH] bump: lomiri

---
 anda/desktops/lomiri/lomiri.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/anda/desktops/lomiri/lomiri.spec b/anda/desktops/lomiri/lomiri.spec
index 235d9d36..154fd3f9 100644
--- a/anda/desktops/lomiri/lomiri.spec
+++ b/anda/desktops/lomiri/lomiri.spec
@@ -1,5 +1,5 @@
 %global forgeurl https://gitlab.com/ubports/development/core/lomiri
-%global commit cb614060a2f76f783ff7651a10e6173d2b70e464
+%global commit 7337620b1563bccf9d82741c88515e32247b2e8f
 %forgemeta
 
 Name:          lomiri
-- 
2.41.0

