--- mise-2026.5.6/Cargo.toml	2006-07-24T01:21:28+00:00
+++ mise-2026.5.6/Cargo.toml	2026-05-13T07:19:23.400768+00:00
@@ -624,31 +624,6 @@
 optional = true
 default-features = false
 
-[target."cfg(windows)".dependencies.self_update]
-version = "0.44"
-features = [
-    "archive-zip",
-    "compression-zip-deflate",
-    "signatures",
-    "reqwest",
-    "rustls",
-]
-optional = true
-default-features = false
-
-[target."cfg(windows)".dependencies.sevenz-rust]
-version = "0.6"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.9"
-features = [
-    "consoleapi",
-    "minwindef",
-]
-
-[target."cfg(windows)".dependencies.zipsign-api]
-version = "0.2"
-
 [lints.clippy]
 borrowed_box = "allow"
 
@@ -663,3 +638,4 @@
 [profile.serious]
 lto = true
 inherits = "release"
+
