--- a/src/main.zig	2025-12-27 19:19:30.000000000 -0600
+++ b/src/main.zig	2025-12-31 08:25:12.962257290 -0600
@@ -3998,6 +3998,15 @@
 
             try create_module.lib_directories.ensureUnusedCapacity(arena, paths.lib_dirs.items.len);
             for (paths.lib_dirs.items) |path| addLibDirectoryWarn2(io, &create_module.lib_directories, path, true);
+
+            for (paths.lib_dirs.items) |native_lib_dir| {
+                for (0.., create_module.rpath_list.items) |i, rpath_dir| {
+                    if (std.mem.eql(u8, native_lib_dir, rpath_dir)) {
+                        _ = create_module.rpath_list.orderedRemove(i);
+                        break;
+                    }
+                }
+            }
         }
 
         if (create_module.libc_paths_file) |paths_file| {
