diff --git a/crates/youki/Cargo.toml b/crates/youki/Cargo.toml
index b494cf1..e415f15 100644
--- a/crates/youki/Cargo.toml
+++ b/crates/youki/Cargo.toml
@@ -49,6 +49,7 @@ wasmtime-wasi = {version = "10.0.2", optional = true }
 tracing = { version = "0.1.37", features = ["attributes"]}
 tracing-subscriber = { version = "0.3.16", features = ["json", "env-filter"] }
 tracing-journald = "0.3.0"
+protobuf = "= 3.2.0"
 
 [dev-dependencies]
 serial_test = "2.0.0"
