--- a/mk/common.mk
+++ b/mk/common.mk
@@ -19,6 +19,7 @@
 BMAKE    ?= MAKEFLAGS= bmake
 DOCKER   ?= docker
 PATCH    ?= patch
+REVISION ?= $(shell git rev-parse HEAD)
 
 UID      := $(shell id -u)
 GID      := $(shell id -g)
@@ -27,7 +28,6 @@
 else
     DATE := $(shell date -u --iso-8601=minutes)
 endif
-REVISION ?= $(shell git rev-parse HEAD)
 COMPILER := $(realpath $(shell which $(CC)))
 PLATFORM ?= $(shell uname -m)
