diff --git a/Cargo.toml b/Cargo.toml
index 9e8224fc..8fbbe54c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -127,7 +127,7 @@ mouce = { git="https://github.com/fufesou/mouce.git" }
 evdev = { git="https://github.com/fufesou/evdev" }
 dbus = "0.9"
 dbus-crossroads = "0.5"
-pam = { git="https://github.com/fufesou/pam", optional = true }
+pam = { path = "./pam", optional = true }
 users = { version = "0.11" }
 
 [target.'cfg(target_os = "android")'.dependencies]
