From db7faffa6d4da9c53a34180854508ad89acf16de Mon Sep 17 00:00:00 2001
From: Cappy Ishihara <cappy@cappuchino.xyz>
Date: Fri, 20 May 2022 19:26:32 +0700
Subject: [PATCH] Remove frozen lock

---
 justfile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/justfile b/justfile
index 826049d..261fb13 100644
--- a/justfile
+++ b/justfile
@@ -2,7 +2,7 @@ debug := '0'
 vendor := '0'
 
 target := if debug == '1' { 'debug' } else { 'release' }
-vendor_args := if vendor == '1' { '--frozen --offline' } else { '' }
+vendor_args := if vendor == '1' { '--offline' } else { '' }
 debug_args := if debug == '1' { '' } else { '--release' }
 cargo_args := vendor_args + ' ' + debug_args
 
-- 
2.36.1

