--- a/Cargo.toml	2025-04-27 10:52:14.000000000 -0500
+++ b/Cargo.toml	2025-04-27 12:39:59.385400385 -0500
@@ -265,22 +265,6 @@
 ]
 default-features = false
 
-[target."cfg(windows)".dependencies.deelevate]
-version = "0.2.0"
-
-[target."cfg(windows)".dependencies.windows]
-version = "0.61.0"
-features = [
-    "Win32_Foundation",
-    "Win32_UI_Shell",
-    "Win32_Security",
-    "Win32_System_Threading",
-    "Win32_Storage_FileSystem",
-]
-
-[target."cfg(windows)".build-dependencies.winres]
-version = "0.1.12"
-
 [profile.bench]
 lto = "thin"
 codegen-units = 16
 [dev-dependencies]
 mockall = "0.13.1"
 tempfile = "3.19.1"
 
