From f26efc2b94f630213cb78d378cd7ba5adbc58260 Mon Sep 17 00:00:00 2001
From: windowsboy111 <wboy111@outlook.com>
Date: Tue, 13 Jun 2023 23:00:14 +0800
Subject: [PATCH] remove libadwaita dependency

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 5766bcf..5d769be 100644
--- a/meson.build
+++ b/meson.build
@@ -13,7 +13,6 @@ base_id = 'app.drey.PaperPlane'
 dependency('glib-2.0', version: '>= 2.72')
 dependency('gio-2.0', version: '>= 2.72')
 dependency('gtk4', version: '>= 4.10')
-dependency('libadwaita-1', version: '>= 1.4')
 dependency('tdjson', version: '== 1.8.14')
 
 glib_compile_resources = find_program('glib-compile-resources', required: true)
-- 
2.40.1

