--- bash_kernel-*/bash_kernel/display.py.orig	2023-11-24 13:17:44.000000000 -0700
+++ bash_kernel-*/bash_kernel/display.py	2024-06-10 11:34:54.360227207 -0600
@@ -52,7 +52,7 @@ display_data_for_<new_type>; (3) Create
 is your friend to debug the format of the content message.
 """
 import base64
-import imghdr
+import filetype
 import json
 import os
 import re
@@ -102,14 +102,17 @@ def display_data_for_image(filename):
         image = f.read()
     _unlink_if_temporary(filename)
 
-    image_type = imghdr.what(None, image)
-    if image_type is None:
+    try:
+        image_type = filetype.guess_mime(image)
+    except TypeError:
+        image_type = None
+    if image_type is None or not image_type.startswith("image/"):
         raise ValueError("Not a valid image: %s" % image)
 
     image_data = base64.b64encode(image).decode('ascii')
     content = {
         'data': {
-            'image/' + image_type: image_data
+            image_type: image_data
         },
         'metadata': {}
     }
--- bash_kernel-*/pyproject.toml.orig	2023-11-24 13:17:44.000000000 -0700
+++ bash_kernel-*/pyproject.toml	2024-06-10 11:28:45.037288031 -0600
@@ -8,7 +8,7 @@ authors = [
     {name = "Thomas Kluyver", email = "thomas@kluyver.me.uk"},
 ]
 readme = "README.rst"
-dependencies = ["pexpect (>=4.0)", "ipykernel"]
+dependencies = ["pexpect (>=4.0)", "ipykernel", "filetype"]
 classifiers = [
     "Framework :: Jupyter",
     "License :: OSI Approved :: BSD License",
