diff --git a/src/util/00-radv-defaults.conf b/src/util/00-radv-defaults.conf
index e36357b9e4a..68a1957ee7b 100644
--- a/src/util/00-radv-defaults.conf
+++ b/src/util/00-radv-defaults.conf
@@ -311,5 +311,9 @@ Application bugs worked around in this file:
         <application name="No Man's Sky" application_name_match="No Man's Sky">
             <option name="radv_app_layer" value="no_mans_sky" />
         </application>
+
+        <application name="Forza Horizon 5" application_name_match="ForzaHorizon5.exe">
+            <option name="vk_x11_override_min_image_count" value="4" />
+        </application>
     </device>
 </driconf>
