diff --git a/Makefile b/Makefile
index 6db2b96..232c993 100644
--- a/Makefile
+++ b/Makefile
@@ -113,7 +113,7 @@ install-data-asusd_user:
 
 .PHONY: install-data-asusd install-data-asusd_user
 
-install-data: install-data-asusd install-data-rog_gui
+install-data: install-data-asusd install-data-rog_gui install-data-asusd_user
 
 install: install-program install-data
 	$(INSTALL_DATA) "./LICENSE" "$(DESTDIR)$(datarootdir)/asusctl/LICENSE"
