--- wine-7.20.old/aclocal.m4	2022-11-17 17:19:01.772386752 +0100
+++ wine-7.20/aclocal.m4	2022-11-17 17:24:03.721683055 +0100
@@ -279,15 +279,17 @@
 dnl
 dnl Usage: WINE_CHECK_DEFINE(name),[action-if-yes,[action-if-no]])
 dnl
 AC_DEFUN([WINE_CHECK_DEFINE],
 [AS_VAR_PUSHDEF([ac_var],[ac_cv_cpp_def_$1])dnl
-AC_CACHE_CHECK([whether we need to define $1],ac_var,
-    AC_EGREP_CPP(yes,[#ifndef $1
-yes
-#endif],
-    [AS_VAR_SET(ac_var,yes)],[AS_VAR_SET(ac_var,no)]))
+AC_CACHE_CHECK([whether we need to define $1], ac_var,
+     [AC_PREPROC_IFELSE([[
+#ifndef $1
+#error not defined
+#endif
+]],
+     [AS_VAR_SET(ac_var,no)],[AS_VAR_SET(ac_var,yes)])])
 AS_VAR_IF([ac_var],[yes],
       [CFLAGS="$CFLAGS -D$1"
   LINTFLAGS="$LINTFLAGS -D$1"])dnl
 AS_VAR_POPDEF([ac_var])])
 
