diff -up gst-plugins-bad-1.18.2/ext/openh264/meson.build.orig gst-plugins-bad-1.18.2/ext/openh264/meson.build
--- gst-plugins-bad-1.18.2/ext/openh264/meson.build.orig	2020-12-06 14:24:13.000000000 +0100
+++ gst-plugins-bad-1.18.2/ext/openh264/meson.build	2021-02-11 11:48:58.660450319 +0100
@@ -4,9 +4,7 @@ openh264_sources = [
   'gstopenh264plugin.c',
 ]
 
-openh264_dep = dependency('openh264', version : '>= 1.3.0',
-                          required : get_option('openh264'),
-                          fallback: ['openh264', 'openh264_dep'])
+openh264_dep = cc.find_library('openh264')
 
 if openh264_dep.found()
   gstopenh264 = library('gstopenh264',
