--- topgrade-16.9.0/Cargo.toml	2006-07-24T01:21:28+00:00
+++ topgrade-16.9.0/Cargo.toml	2026-02-13T07:55:20.141538+00:00
@@ -255,29 +255,6 @@
 default-features = false
 package = "self_update"
 
-[target."cfg(windows)".dependencies.is_elevated]
-version = "~0.1"
-
-[target."cfg(windows)".dependencies.parselnk]
-version = "~0.1"
-
-[target."cfg(windows)".dependencies.self_update_crate]
-version = "~0.42"
-features = [
-    "archive-zip",
-    "compression-zip-deflate",
-    "rustls",
-]
-optional = true
-default-features = false
-package = "self_update"
-
-[target."cfg(windows)".dependencies.windows]
-version = "~0.62"
-features = ["Win32_System_Console"]
-
-[target."cfg(windows)".dependencies.windows-registry]
-version = "~0.6"
-
 [profile.release]
 lto = true
+
