From fdb5a9b177e32b69b0cd72e5481c71612ede2ba1 Mon Sep 17 00:00:00 2001
From: Olivia <git@olivia.sh>
Date: Sun, 19 Jul 2026 11:43:42 -0500
Subject: [PATCH] Install manpage to correct location

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 76d6145..aae5b23 100644
--- a/Makefile
+++ b/Makefile
@@ -146,11 +146,11 @@ $(BINARY): $(<<objects>>)
 
 install: $(BINARY)
 	install -Dm0755 $(BINARY) $(DESTDIR)$(PREFIX)/$(BINDIR)/$(BINARY)
-	install -Dm0644 activate-linux.1 $(MANDIR)/man1/activate-linux.1
+	install -Dm0644 activate-linux.1 $(DESTDIR)$(MANDIR)/man1/activate-linux.1
 
 uninstall:
 	sudo $(RM) -f $(DESTDIR)$(PREFIX)/$(BINDIR)/$(BINARY)
-	sudo $(RM) -f $(MANDIR)/man1/activate-linux.1
+	sudo $(RM) -f $(DESTDIR)$(MANDIR)/man1/activate-linux.1
 	sudo mandb -q
 
 appimage: $(BINARY)
-- 
2.55.0

