--- oxipng-9.1.2/Cargo.toml	1970-01-01T00:00:01+00:00
+++ oxipng-9.1.2/Cargo.toml	2024-07-13T07:08:09.478330+00:00
@@ -150,7 +150,6 @@
 [features]
 binary = [
     "dep:clap",
-    "dep:glob",
     "dep:env_logger",
 ]
 default = [
@@ -169,13 +168,10 @@
 sanity-checks = ["dep:image"]
 zopfli = ["dep:zopfli"]
 
-[target."cfg(windows)".dependencies.glob]
-version = "0.3.1"
-optional = true
-
 [badges.maintenance]
 status = "actively-developed"
 
 [badges.travis-ci]
 branch = "master"
 repository = "shssoichiro/oxipng"
+
