diff --git a/ext/lcevcdecoder/gstlcevcdec.c b/ext/lcevcdecoder/gstlcevcdec.c
index a1af5b11390b..3c9cb2ffdf2b 100644
--- a/ext/lcevcdecoder/gstlcevcdec.c
+++ b/ext/lcevcdecoder/gstlcevcdec.c
@@ -539,7 +539,7 @@ send_enhancement_data (GstLcevcDec * lcevc, GstBuffer * input_buffer)
   }
 
   if (LCEVC_SendDecoderEnhancementData (lcevc->decoder_handle,
-          input_buffer->pts, TRUE, enhancement_info.data,
+          input_buffer->pts, enhancement_info.data,
           enhancement_info.size) != LCEVC_Success) {
     GST_INFO_OBJECT (lcevc,
         "Could not send input buffer %" GST_TIME_FORMAT
@@ -581,7 +581,7 @@ send_base_picture (GstLcevcDec * lcevc, GstBuffer * input_buffer)
     goto done;
   }
 
-  if (LCEVC_SendDecoderBase (lcevc->decoder_handle, input_buffer->pts, TRUE,
+  if (LCEVC_SendDecoderBase (lcevc->decoder_handle, input_buffer->pts,
           picture_handle, 1000000, NULL) != LCEVC_Success) {
     GST_ELEMENT_ERROR (lcevc, STREAM, DECODE, (NULL),
         ("Could not send input buffer %" GST_TIME_FORMAT " base picture",
diff --git a/ext/lcevcdecoder/meson.build b/ext/lcevcdecoder/meson.build
index ecc5be034a95..ead6b9e9a361 100644
--- a/ext/lcevcdecoder/meson.build
+++ b/ext/lcevcdecoder/meson.build
@@ -26,7 +26,7 @@ plugin_sources += {
   'lcevcdecoder': pathsep.join(doc_sources)
 }
 
-lcevc_dec_dep = dependency ('lcevc_dec', required: get_option('lcevcdecoder'))
+lcevc_dec_dep = dependency ('lcevc_dec', version: '>= 4', required: get_option('lcevcdecoder'))
 
 if lcevc_dec_dep.found()
   gstlcevcdecoder = library('gstlcevcdecoder',
