diff --git a/data/raindrop.desktop.in b/data/raindrop.desktop.in
index 69e66f7..a042659 100644
--- data/raindrop.desktop.in
+++ data/raindrop.desktop.in
@@ -3,7 +3,7 @@ Type=Application
 Name=Screen Configuration
 Comment=Configure monitor arrangement, scaling and orientation
 Icon=computer
-Exec=raindrop
+Exec=sh -c 'pkexec /usr/bin/raindrop'
 Terminal=false
 Categories=Settings;
 StartupNotify=true
