diff --git a/setup.py b/setup.py
index 3eb7110..24c73f9 100644
--- a/setup.py
+++ b/setup.py
@@ -49,7 +49,7 @@ setup(name="magic-wormhole",
           "spake2==0.9", "pynacl",
           "attrs >= 19.2.0", # 19.2.0 replaces cmp parameter with eq/order
           "twisted[tls] >= 17.5.0", # 17.5.0 adds failAfterFailures=
-          "autobahn[twisted] >= 0.14.1, != 25.9.1, != 25.10.1, != 25.10.2",
+          "autobahn[twisted] >= 0.14.1, != 25.9.1, != 25.10.1",
           "automat",
           "cryptography",
           "tqdm >= 4.13.0", # 4.13.0 fixes crash on NetBSD
