--- mise-2026.2.10/Cargo.toml	2006-07-24T01:21:28+00:00
+++ mise-2026.2.10/Cargo.toml	2026-02-12T15:02:18.584012+00:00
@@ -575,30 +575,6 @@
 optional = true
 default-features = false
 
-[target."cfg(windows)".dependencies.self_update]
-version = "0.42"
-features = [
-    "archive-zip",
-    "compression-zip-deflate",
-    "signatures",
-    "rustls",
-]
-optional = true
-default-features = false
-
-[target."cfg(windows)".dependencies.sevenz-rust]
-version = "0.6"
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.9"
-features = [
-    "consoleapi",
-    "minwindef",
-]
-
-[target."cfg(windows)".dependencies.zipsign-api]
-version = "0.1"
-
 [lints.clippy]
 borrowed_box = "allow"
 
@@ -615,3 +591,4 @@
 [profile.serious]
 lto = true
 inherits = "release"
+
