diff --git a/test/cases.zig b/test/cases.zig
index 7b4d78d..845f0ce 100644
--- a/test/cases.zig
+++ b/test/cases.zig
@@ -55,7 +55,7 @@ pub fn lowerCases(
     };
 
     const test_case_dir = b.build_root.join(b.allocator, &.{"test/cases"}) catch unreachable;
-    var dir = std.Io.Dir.openDirAbsolute(io, test_case_dir, .{ .iterate = true }) catch |err| {
+    var dir = b.build_root.handle.openDir(io, test_case_dir, .{ .iterate = true }) catch |err| {
         const fail_step = b.addFail(b.fmt("unable to open test/cases: {s}", .{@errorName(err)}));
         test_translate_step.dependOn(&fail_step.step);
         test_run_translated_step.dependOn(&fail_step.step);
