From 55db3aeaeb962952881f73e94432f750cfb64fc8 Mon Sep 17 00:00:00 2001
From: Jan200101 <sentrycraft123@gmail.com>
Date: Thu, 15 Feb 2024 16:24:00 +0100
Subject: [PATCH 09/10] main: use Discrete key in switcherooctl

Signed-off-by: Jan200101 <sentrycraft123@gmail.com>
---
 src/switcherooctl.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/switcherooctl.in b/src/switcherooctl.in
index 96c21cc..c0e3f07 100755
--- a/src/switcherooctl.in
+++ b/src/switcherooctl.in
@@ -77,6 +77,7 @@ def print_gpu(gpu, index):
     print('Device:', index)
     print('  Name:       ', gpu['Name'])
     print('  Default:    ', "yes" if gpu['Default'] else "no")
+    print('  Discrete:   ', "yes" if gpu['Discrete'] else "no")
     print('  Environment:', env_to_str(gpu['Environment']))
 
 def _list():
@@ -126,7 +127,7 @@ def get_discrete_gpu():
         return None
 
     try:
-        gpu = next(gpu for gpu in gpus if not gpu['Default'])
+        gpu = next(gpu for gpu in gpus if gpu['Discrete'])
     except StopIteration:
         return None
     else:
-- 
2.46.0

