--- television-0.14.4/Cargo.toml	1970-01-01T00:00:01+00:00
+++ television-0.14.4/Cargo.toml	2026-01-04T18:45:00.790925+00:00
@@ -251,19 +251,6 @@
 [target.'cfg(not(target_os = "macos"))'.dependencies.crossterm]
 version = "0.28"
 features = ["serde"]
-
-[target.'cfg(target_os = "macos")'.dependencies.crossterm]
-version = "0.28.1"
-features = [
-    "serde",
-    "use-dev-tty",
-]
-
-[target."cfg(windows)".dependencies.clipboard-win]
-version = "5.4.0"
-
-[target."cfg(windows)".dependencies.winapi-util]
-version = "0.1.9"
 
 [lints.clippy]
 cast_precision_loss = "allow"
@@ -301,3 +288,4 @@
 opt-level = 3
 lto = false
 inherits = "release"
+
