From 984b0e50788ed9710d093aa6917604eb2d01c293 Mon Sep 17 00:00:00 2001
From: David Hewitt <davidmhewitt@gmail.com>
Date: Fri, 1 Oct 2021 13:45:29 +0100
Subject: [PATCH] MediaPlayer: Give type field valid name

---
 src/Services/MediaPlayer.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Services/MediaPlayer.vala b/src/Services/MediaPlayer.vala
index 7e0e093..443f930 100644
--- a/src/Services/MediaPlayer.vala
+++ b/src/Services/MediaPlayer.vala
@@ -30,7 +30,7 @@ public interface Sound.Services.MediaPlayer : Object {
 
     public abstract string name { owned get; }
     [DBus (name = "Type")]
-    public abstract string _type { owned get; }
+    public abstract string mediaplayer_type { owned get; }
     public abstract string subtype { owned get; }
     public abstract uint position { get; }
     public abstract string status { owned get; }
