From 6a4ec827b0324c8e7a92f2621944697d311d4533 Mon Sep 17 00:00:00 2001
From: Fabio Valentini <decathorpe@gmail.com>
Date: Wed, 13 Jul 2022 13:58:46 +0200
Subject: [PATCH] meson: remove deprecated positional arguments from
 i18n.merge_file calls

---
 data/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 899be41..4afaf03 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -27,7 +27,6 @@ install_data(
 
 i18n = import('i18n')
 desktop_file = i18n.merge_file(
-    'desktop',
     input: meson.project_name() + '.desktop.in',
     output: meson.project_name() + '.desktop',
     install: true,
@@ -37,7 +36,6 @@ desktop_file = i18n.merge_file(
 )
 
 appdata_file = i18n.merge_file(
-    'appdata',
     input: meson.project_name() + '.appdata.xml.in',
     output: meson.project_name() + '.appdata.xml',
     install: true,
-- 
2.37.1

