diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 5a4ecd7f..b78a436f 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -16,8 +16,8 @@ include_directories("${GTEST_SOURCE_DIR}/googletest/include" "${GTEST_SOURCE_DIR
 
 # coverage
 # https://gcovr.com/en/stable/guide/compiling.html#compiler-options
-set(CMAKE_CXX_FLAGS "-fprofile-arcs -ftest-coverage -ggdb -O0")
-set(CMAKE_C_FLAGS "-fprofile-arcs -ftest-coverage -ggdb -O0")
+set(CMAKE_CXX_FLAGS "-fprofile-arcs -ftest-coverage -ggdb -O0 -fPIE")
+set(CMAKE_C_FLAGS "-fprofile-arcs -ftest-coverage -ggdb -O0 -fPIE")
 
 # if windows
 if (WIN32)
