From 57d451554d68767b7568900b88d2b9fc46b36a46 Mon Sep 17 00:00:00 2001
From: Sergey Avseyev <sergey.avseyev@gmail.com>
Date: Fri, 6 Oct 2017 23:08:16 +0300
Subject: [PATCH 2/5] use _datadir for platform-independent library path

---
 compiler/options.nim | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler/options.nim b/compiler/options.nim
index 45ee48944..ee1ac7363 100644
--- a/compiler/options.nim
+++ b/compiler/options.nim
@@ -507,7 +507,7 @@ proc setDefaultLibpath*(conf: ConfigRef) =
     var prefix = getPrefixDir(conf)
     when defined(posix):
       if prefix == AbsoluteDir"/usr":
-        conf.libpath = AbsoluteDir"/usr/lib/nim"
+        conf.libpath = AbsoluteDir"/usr/share/nim"
       elif prefix == AbsoluteDir"/usr/local":
         conf.libpath = AbsoluteDir"/usr/local/lib/nim"
       else:
-- 
2.23.0

