--- a/desktop/packages/mullvad-vpn/tasks/distribution.js	2025-11-27 10:32:49.229089406 -0600
+++ b/desktop/packages/mullvad-vpn/tasks/distribution.js	2025-11-27 10:54:15.873776987 -0600
@@ -178,16 +178,6 @@
     },
 
     linux: {
-      target: [
-        {
-          target: 'deb',
-          arch: getLinuxTargetArch(),
-        },
-        {
-          target: 'rpm',
-          arch: getLinuxTargetArch(),
-        },
-      ],
       executableName: 'mullvad-vpn',
       artifactName: 'MullvadVPN-${version}_${arch}.${ext}',
       category: 'Network',
