diff -Naur LCEVCdec-3.3.5.old/docs/sphinx/CMakeLists.txt LCEVCdec-3.3.5/docs/sphinx/CMakeLists.txt
--- LCEVCdec-3.3.5.old/docs/sphinx/CMakeLists.txt	2025-03-23 16:27:09.553289475 +0100
+++ LCEVCdec-3.3.5/docs/sphinx/CMakeLists.txt	2025-03-24 08:54:32.562734384 +0100
@@ -41,8 +41,8 @@
     documentation ALL
     DEPENDS doxygen
     COMMAND
-        ${SPHINX_EXECUTABLE} -b html -W -n -a -Dversion=${GIT_SHORT_VERSION}
+        ${SPHINX_EXECUTABLE} ${SPHINX_SOURCE} ${SPHINX_BUILD}
+        -b html -W -n -a -Dversion=${GIT_SHORT_VERSION}
         -Drelease=${GIT_VERSION} -Dbreathe_projects.LCEVC_DEC=${DOXYGEN_OUTPUT_DIR}/xml
-        ${SPHINX_SOURCE} ${SPHINX_BUILD}
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     COMMENT "Generating docs with Sphinx")
