diff -Naur LCEVCdec-3.3.5.old/src/api/src/buffer_manager.h LCEVCdec-3.3.5/src/api/src/buffer_manager.h
--- LCEVCdec-3.3.5.old/src/api/src/buffer_manager.h	2025-03-22 11:46:05.946962992 +0100
+++ LCEVCdec-3.3.5/src/api/src/buffer_manager.h	2025-03-22 11:46:36.029979290 +0100
@@ -15,6 +15,7 @@
 #ifndef VN_API_BUFFER_MANAGER_H_
 #define VN_API_BUFFER_MANAGER_H_
 
+#include <cstdint>
 #include <functional>
 #include <memory>
 #include <set>
